yaml/semgrep/missing-category.yaml

```diff
@@ -0,0 +1,20 @@
+rules:
+- id: missing-category
+  message: >-
+    This Semgrep rule is missing a valid 'category' field in the 'metadata'.
+    'category' must be one of 'security', 'correctness', 'best-practice',
+    'performance', 'maintainability', or 'portability'.
+  severity: INFO
+  patterns:
+  - pattern-inside: 'rules: [..., $RULE, ...]'
+  - pattern: $RULE
+  - pattern-not: |
+      ...
```

Comment:
Do we need this `...`? I know hanging `...` at the beginning or end tend to slow things down.

Reply:
Good question, I didn't test that! brb

Reply:
Not needed, good catch 👍